### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by samdealy**
### Motivation
As described in the [ANDROID-67 ticket](https://mobilecoin.atlassian.net/browse/ANDROID-67), we're working on updating existing and adding new unit tests. Our goal is to improve readability and correctness. 

### In this PR
This PR mainly updates the UTXOSelectorTest. The test method signatures were renamed in the form <METHOD_NAME>_<INPUT_INFORMATION>_<EXPECTED_OUTCOME>. I think this helps the reader better understand what to look for when evaluating the test. The method signatures `throws` declarations no longer list each specific error that the method under test can throw, and instead lists `Exception`. This helps improve readability because the specific exceptions that can be thrown are normally not germane to the test itself. Since we test for specific errors in these unit tests, we can be sure that exceptions are being thrown as expected. 

I also added a few new unit tests that mostly test when an input is null or empty.

Another change I made in both UTXOSelector and UTXOSelectorTest is to change a variable's type to its most general interface possible. For example, instead of typing a variable as `ArrayList`, it's now typed as `List`. This helps the reader understand that only the `List` methods are needed for this object.  Note that I left the `Vector` type as is because we use a method that is specific to Vector and not defined in a more abstract class. See Effective Java Item 22 : "Use interfaces only to define types" for more info.

### Future Work
Continue working on modifying and updating unit tests.

### CODE REVIEW

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/UTXOSelectorTest.java [hunk 1]
```diff
@@ -14,253 +14,394 @@
...
-                BigInteger.ZERO,
-                2
-        );
+    thrown.expect(InsufficientFundsException.class);
+    UTXOSelector.selectTxOutNodesForMerging(utxos, txFee, inputFee, /* outputFee=
+     */ BigInteger.ZERO);
+  }
+
+  @Test
+  public void selectTxOutNodesForMerging_amountCoversFees_twoMerges_selectsCorrectNodesBothTimes() throws Exception {
```
[the remark this answers is not in the document]
Comment (samdealy) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/UTXOSelectorTest.java]: Done.

**Review by voloshyn — APPROVED** [review]
